stan/math/prim/arr/functor/integrate_ode_rk45.hpp

```diff
@@ -127,7 +127,7 @@ namespace stan {
                                                            double,
                                                            std::vector<double>,
                                                            double>() ),
-                      coupled_system,
+                      boost::ref(coupled_system),
```

Comment:
Hi!

odeint does lots of copy-constructions of the ode RHS functor. The author of odeint claims that with compiler optimizations this will make sure that code being generated is faster. However, I have observed the reverse. That is, since we have some instances as copies inside the ode rhs functor, lots of copies are being made of these during the ode integration. I noticed a few % speedup when using boost::ref. So I am not saving a single, but many copies.

Sebastian